spectre-meltdown-checker.sh

```diff
@@ -41,16 +41,20 @@ check_vmlinux()
 
 try_decompress()
 {
-        # The obscure use of the "tr" filter is to work around older versions of
-        # "grep" that report the byte offset of the line instead of the pattern.
+	# The obscure use of the "tr" filter is to work around older versions of
+	# "grep" that report the byte offset of the line instead of the pattern.
 
-        # Try to find the header ($1) and decompress from here
-        for     pos in `tr "$1\n$2" "\n$2=" < "$4" | grep -abo "^$2"`
-        do
-                pos=${pos%%:*}
-                tail -c+$pos "$4" | $3 > $vmlinuxtmp 2> /dev/null
-                check_vmlinux "$vmlinuxtmp" && echo "$vmlinuxtmp" && return 0
-        done
+	# Try to find the header ($1) and decompress from here
+	for     pos in `tr "$1\n$2" "\n$2=" < "$5" | grep -abo "^$2"`
+	do
+		if ! which $3 >/dev/null 2>&1; then
+			echo "missing '$3' tool, please install it, usually it's in the '$4' package" > /dev/stderr
```

Comment:
To keep the extract-vmlinux functions silent and use those like a library, could you replace this by the definition of a variable such as vmlinux_extract_error, and test for it's emptiness after the extract-vmlinux func call for example ?